astropy/table/table.py

```diff
@@ -799,18 +803,21 @@ def __init__(self, data=None, masked=False, names=None, dtype=None,
         elif data is None:
             if names is None:
                 if dtype is None:
-                    if meta is not None:
-                        self.meta = deepcopy(meta) if copy else meta
```

Comment:
This was previously repeating code from below and then jumping out early, making the code harder to follow.